### PR TITLE
Adding possibility to supply the jwt private key as a string

### LINKED ISSUE #40078: Support JWT `key ` property for the OIDC client authentication
### Discussed in https://github.com/quarkusio/quarkus/discussions/40036

Both `quarkus-oidc-client` and `quarkus-oidc` need to talk to the OIDC provider and they must authenticate to the OIDC provider in some cases. Many authentication options are supported such as the client id and secret, but also several JWT options, where a JWT is generated to support the client authentication.
This JWT can be signed with the client secret. Or with the private key - and this private key must currently be located in the PEM file or the keystore.

However it would be great if it were also possible to supply the private key as a string so that it can inferred from various environment properties, without having to create files

### PR DESCRIPTION
**Pull request by IvanPuntev**
Fixes #40078

### CODE REVIEW

**Comment by IvanPuntev** [comment]
Related github issue - https://github.com/quarkusio/quarkus/issues/40078

**Comment by sberyozkin** [comment]
@IvanPuntev Thanks, please see https://github.com/quarkusio/quarkus/pull/40085#issuecomment-2057346452 on how to add the test,
also add one more example for the `private_key_jwt` method at the end of https://quarkus.io/guides/security-openid-connect-client-reference#oidc-client-authentication, and similarly, at https://quarkus.io/guides/security-oidc-code-flow-authentication#oidc-provider-client-authentication (the code you updated is shared between the two extensions)

File: extensions/oidc-common/runtime/src/main/java/io/quarkus/oidc/common/runtime/OidcCommonConfig.java [hunk 1]
```diff
@@ -274,6 +274,13 @@ public static enum Source {
             @ConfigItem
             public Provider secretProvider = new Provider();
 
+            /**
+             * The string representation of the private key content no matter including
```
Comment (sberyozkin) [inline extensions/oidc-common/runtime/src/main/java/io/quarkus/oidc/common/runtime/OidcCommonConfig.java:278]:
```suggestion
             * The string representation of the private key content in PEM format.
```

File: extensions/oidc-common/runtime/src/main/java/io/quarkus/oidc/common/runtime/OidcCommonConfig.java [hunk 2]
```diff
@@ -274,6 +274,13 @@ public static enum Source {
...
+             * or excluding the ----- PRIVATE KEY / CERTIFICATE ----- prefix and postfix.
```
Comment (sberyozkin) [inline extensions/oidc-common/runtime/src/main/java/io/quarkus/oidc/common/runtime/OidcCommonConfig.java:279]:
```suggestion
             * The 'PRIVATE KEY' delimiters are not required.
```

Comment (sberyozkin) [reply extensions/oidc-common/runtime/src/main/java/io/quarkus/oidc/common/runtime/OidcCommonConfig.java:279]: Certificate one would contain the public key...

**Comment by sberyozkin** [comment]
@IvanPuntev Ping me please if you need the help with tests or docs - we already have Keycloak container setup to expect this type of authentication, so with those minor updates to the test it should be all good

File: extensions/oidc-common/runtime/src/main/java/io/quarkus/oidc/common/runtime/OidcCommonConfig.java (hunk 2 above)
Comment (sberyozkin) [reply extensions/oidc-common/runtime/src/main/java/io/quarkus/oidc/common/runtime/OidcCommonConfig.java:279]: @IvanPuntev Actually, can you please drop the whole line, as inlined JWK format is also supported with that function call

**Comment by IvanPuntev** [comment]
@sberyozkin Is this what is needed for the test? https://github.com/quarkusio/quarkus/pull/40086/commits/f594025283d971fda8efff95cb8c670e5bc82e0d
Also I need a hint where to edit the guides.

**Comment by sberyozkin** [comment]
@IvanPuntev It looks great, thanks, let me only ask you to align a little bit the JavaDoc for this property with other properties, I'll suggest something, and then please squash the  commits and it will be ready to go

File: extensions/oidc-common/runtime/src/main/java/io/quarkus/oidc/common/runtime/OidcCommonConfig.java [hunk 3]
```diff
@@ -274,6 +274,12 @@ public static enum Source {
             @ConfigItem
             public Provider secretProvider = new Provider();
 
+            /**
+             * The string representation of the private key content.
```
Comment (sberyozkin) [inline extensions/oidc-common/runtime/src/main/java/io/quarkus/oidc/common/runtime/OidcCommonConfig.java:278]:
```suggestion
             * String representation of a private key. If provided, indicates that JWT is signed using a private key in PEM or JWK format.             
```

As far as I recall, the Doc team suggested `If provided, ` and this is also similar how other properties are documented.

And then add one more doc line, '*You can use the {@link #signatureAlgorithm} property to override the default key algorithm, `RS256`.'

This is the last suggestion, please squash afterwards and open the PR for review for the builds to run

Thanks

**Comment by sberyozkin** [comment]
@IvanPuntev Oh, sorry, forgot about the guides, they are here

https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc

I can push the update to your branch if you'd like, let me know

**Comment by IvanPuntev** [comment]
@sberyozkin Does it look good? https://github.com/quarkusio/quarkus/pull/40086/commits/78315a9c166606047fd3f163e5b2f9ebcbeefeaf If everything is ok then I will do the squash.

**Comment by sberyozkin** [comment]
@IvanPuntev Thanks for that, yes, looks good, I've proposed minor updates

**Comment by sberyozkin** [comment]
@IvanPuntev Minor formatting issue, I'm fixing it right now

**Review by sberyozkin — APPROVED** [review]

**Pull request merged**